5.0/en/0x12-V3-Session-management.md

```diff
@@ -91,10 +90,88 @@ Attackers start a half-open attack by attempting to lock, reset, or recover a cr
+
+## Terminology: 
+
+* `Access tokens` --- provides an abstraction, replacing different authorization constructs (e.g., username and password, assertion) for a single token understood by the resource server. This abstraction enables issuing access tokens valid for a short time period, as well as removing the resource server's need to understand a wide range of authentication schemes. 
+* `Refresh tokens` --- are credentials used to obtain access tokens. These are issued to the client by the authorization server and are used to obtain a new access token when the current access token becomes invalid or expires, or to obtain additional access tokens with identical or narrower scope (access tokens may have a shorter lifetime and fewer permissions than authorized by the resource owner). 
+* `Client` --- generally refers to an application making protected resource requests on behalf of the resource owner and with its authorization. The term "client" does not imply any particular implementation characteristics (e.g., whether the application executes on a server, a desktop, or other devices). 
+* `Authorization Server (AS)` --- refers to the server issuing access tokens to the client after successfully authenticating the resource owner and obtaining authorization.
+* `Resource Owner (RO)` --- refers to an entity capable of granting access to a protected resource. When the resource owner is a person, it is referred to as an end-user. 
+* `Resource Server (RS)` --- refers to the server hosting the protected resources, capable of accepting and responding to protected resource requests using access tokens. 
+
+
+## OAuth 2.0 Essential Basics
+
+| # | Description | L1 | L2 | L3 |
+| :---: | :--- | :---: | :---: | :---: |
+| **3.8.1** | [ADDED] Verify that Clients do not expose URLs that forward the user's browser to arbitrary URIs obtained from a query parameter ("open redirectors") which can enable exfiltration of authorization codes and access tokens.  | ✓ | ✓ | ✓ |
+| **3.8.2** | [ADDED] Verify Clients have ensured that the Authorization Server supports PKCE may rely on the CRSF protection provided by PKCE. In OpenID Connect flows, the "nonce" parameter provides CSRF protection. Otherwise, one-time user CSRF tokens carried in the "state" parameter that are securely bound to the user agent must be used for CSRF protection.  | ✓ | ✓ | ✓ |
+| **3.8.3** | [ADDED] Verify that when an OAuth Client can interact with more than one Authorization Server, Clients should use the issuer "iss" parameter as a countermeasure, or alternatively based on an "iss" value in the authorization response (such as the "iss" Claim in the ID Token in OpenID) | ✓ | ✓ | ✓ |
+| **3.8.4** | [ADDED] Verify that when the other countermeasure options for OAuth clients interacting with more than one Authorization Servers are absent, Clients may instead use distinct redirect URIs to identify authorization endpoints and token endpoints.  | ✓ | ✓ | ✓ |
```

Comment:
It was actually not so clear what the RFC means by that "other countermeasure options" as well. I would dig deeper into this so that we can come to a conclusion whether to merge this to 3.8.3 or just clarify the wording to make it clearer.

Reply:
Upon reading further, it seems like it makes sense for me to combine **3.8.3** and **3.8.4** into one. I was worried before that it will be a very wordy and long requirement (which we know leads to some parts being omitted when people consider the verification requirement. I have a plan to make it a good one, stay tuned for the new commit.